File: PRÁCTICA SEGUNDO PARCIAL/Ejercicio6/Java/ContarSignosExclamacion.java
import java.util.Scanner;

public class ContarSignosExclamacion {
    public static int contarSignosExclamacion(String cadena) {
        int contador = 0;
        for (char caracter : cadena.toCharArray()) {
            if (caracter == '!') {
                contador++;
            }
        }
        return contador;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese una cadena: ");
        String cadena = scanner.nextLine();
        int cantidadSignos = contarSignosExclamacion(cadena);
        System.out.println("La cadena tiene " + cantidadSignos + " signos de exclamación.");
        scanner.close();
    }
}

File: PRÁCTICA SEGUNDO PARCIAL/Ejercicio6/Python/ContarSignosExclamacion.py
def contar_signos_exclamacion(cadena):
    contador = 0
    for caracter in cadena:
        if caracter == '!':
            contador += 1
    return contador

cadena = input("Ingrese una cadena: ")
cantidad_signos = contar_signos_exclamacion(cadena)
print("La cadena tiene", cantidad_signos, "signos de exclamación.")
